context/src/main/java/io/opentelemetry/context/ContextStorage.java

```diff
@@ -69,6 +69,15 @@ static ContextStorage get() {
     return LazyStorage.get();
   }
 
+  /**
+   * Returns the default {@link ContextStorage} used to attach {@link Context}s to scopes of
+   * execution. Should only be used when defining your own {@link ContextStorage} in case you want
+   * to delegate functionality to the default implementation.
+   */
+  static ContextStorage threadLocalStorage() {
```

Comment:
This makes the method public. What is the reason to do it?

Reply:
To make it easier to write a delegating `ContextStorage` implementation that uses thread-local storage.  This is already suggested by the Javadocs on this type which reference the method `Context#threadLocalStorage` which does not exist.

Reply:
What about then we make the implementation thread local public enum? I feel that exposing this in the Context ties too much the threadlocal storage to the context.

Reply:
That'd be fine by me.  I was going with the existing convention suggested in the Javadocs on `ContextStorage` as well as by `LazyStorage` (which references a similar helper method on `DefaultContext`).  Should we get others to weigh in before I change this PR?

Reply:
Yes, I would like to hear others opinions @anuraaga @jkwatson @carlosalberto